### PR TITLE
LFVM: Add tests for code related context calls

### PR DESCRIPTION
**Pull request by LuisPH3**
part of #751 

This pr adds tests to check parameters passed to context calls for `GetCode`  and  `GetCodeSize`

### CODE REVIEW

File: go/interpreter/lfvm/instructions_test.go [hunk 1]
```diff
@@ -1796,6 +1796,59 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
...
+	runContext.EXPECT().GetCode(address).Return(code)
+
+	ctxt := getEmptyContext()
+	ctxt.context = runContext
+	ctxt.gas = 1 << 32
+
+	ctxt.stack.push(uint256.NewInt(size))   // length
+	ctxt.stack.push(uint256.NewInt(offset)) // codeOffset
+	ctxt.stack.push(uint256.NewInt(0))      // memOffset
+	ctxt.stack.push(new(uint256.Int).SetBytes(address[:]))
```
Comment (facuMH) [inline go/interpreter/lfvm/instructions_test.go:1817]: use fillstack?

**Review by facuMH — APPROVED** [review]

**Review by HerbertJordan — DISMISSED** [review]
nice, thanks!

File: go/interpreter/lfvm/instructions_test.go [hunk 2]
```diff
@@ -1796,6 +1796,61 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
...
+	_, _ = rand.Read(address[:])
+	var offset uint64 = 1
+	var size uint64 = 2
+
+	runContext := tosca.NewMockRunContext(gomock.NewController(t))
+	runContext.EXPECT().GetCode(address).Return(code)
+
+	ctxt := getEmptyContext()
+	ctxt.context = runContext
+	ctxt.gas = 1 << 32
```
Comment (facuMH) [inline go/interpreter/lfvm/instructions_test.go:1812]: unnecessary change, can be removed.

File: go/interpreter/lfvm/instructions_test.go [hunk 3]
```diff
@@ -1796,6 +1796,61 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
...
+func TestInstructions_opExtcodesize_CallsContextAndWritesResultInStack(t *testing.T) {
+
+	address := tosca.Address{}
+	_, _ = rand.Read(address[:])
+	runContext := tosca.NewMockRunContext(gomock.NewController(t))
+	runContext.EXPECT().GetCodeSize(address).Return(1234)
+
+	ctxt := getEmptyContext()
+	ctxt.context = runContext
+	ctxt.gas = 1 << 32
```
Comment (facuMH) [inline go/interpreter/lfvm/instructions_test.go:1840]: same here, you don't need that change. It can confusing because `getEmptyContext` does not generate an empty context but a prefilled one, with more than enough gas.

**Review by facuMH — APPROVED** [review]
looks good, just a couple of nits.

File: go/interpreter/lfvm/instructions_test.go (hunk 3 above)
Comment (LuisPH3) [reply go/interpreter/lfvm/instructions_test.go]: maybe we should rename empty context to test context. I always thought an empty context would have no gas.
